changelog.d/4889.removal

```diff
@@ -0,0 +1 @@
+Remove deprecated experimental restricted space lab option
```

Comment:
FYI: using `removal` extension is reserved for changes in the SDK. For this change `misc` would have been better. (I am seeing that when doing the release, so I will update the changelog manaully, no worries)